fix: Discard minidumps with multipart form data [NATIVE-431]

loewenheim opened the pull request:
I'm not entirely clear if the error we return here would be reported to sentry.

#skip-changelog

flub commented on crates/symbolicator/src/services/symbolication.rs:
```diff
@@ -2118,6 +2118,12 @@ impl SymbolicationActor {
+        // check if the minidump starts with multipart form data and discard it if so
+        let minidump_path = minidump_file.to_path_buf();
+        let minidump = ByteView::open(minidump_path).unwrap_or_else(|_| ByteView::from_slice(b""));
+        if minidump.starts_with(b"--") {
+            return Err(anyhow::anyhow!("Minidump contains multipart form data").into());
+        }
```
I don't think we want to report the error to sentry at all, IIRC the idea of this PR is to get rid of the nightly error spikes we get currently because of a misbehaving user because it results in alert noise.

Instead we would like this specific case to be handled silently *and* emit a metric for it, so we can still observe it happening.

flub reviewed (changes requested):
so i think this is doing the wrong thing, we already get errors we should turn them into metrics.

Also this is meant to specifically check for the multipart case only.  the magic bytes are already checked when we parse the minidump.  We only want to surpress the normal errors if someone (electron) sent a multipart.

loewenheim commented:
I moved the logic outward as far as possible, to `handle_minidump_request`. It now returns `Ok(SymbolicationResponse::Failed)` in case of a minidump with multipart form data (it needs to return something, and from what I can tell, we shouldn't return an `Err`). I also added a metric.

flub reviewed (commented):
I think this is the right place and right way to go about this.  Only question remaining is what is really the desired return value?  I think an OK with failed json inside it is more of a processing error, while really for symbolicator this is a bad request.

I think the code on sentry-side BAD_REQUEST would end up in this line https://github.com/getsentry/sentry/blob/4cf958535563a6b7443b961eac02ea161b3df095/src/sentry/lang/native/symbolicator.py#L600 so would also avoid creating an error.
(while that handling also suggests it doesn't matter, BAD_REQUEST still seems more suitable unless I have HTTP wrong?)

loewenheim commented on crates/symbolicator/src/endpoints/minidump.rs:
```diff
@@ -53,6 +54,15 @@ pub async fn handle_minidump_request(
+    let minidump = ByteView::open(minidump_path).unwrap_or_else(|_| ByteView::from_slice(b""));
+    if minidump.starts_with(b"--") {
+        metric!(counter("symbolication.minidump.multipart_form_data") += 1);
+        return Ok(Json(SymbolicationResponse::Failed {
+            message: "minidump file contains multipart form data".to_string(),
+        }));
```
You mean use the status code as the message in `Failed`?
